### PR TITLE
weblate: address additional review on package and dependencies

### PR DESCRIPTION
**Pull request by erictapen**
## Description of changes

Addresses @dotlambda s reviews in https://github.com/NixOS/nixpkgs/pull/325541#issuecomment-2316564000 and https://github.com/NixOS/nixpkgs/pull/334849#pullrequestreview-2267536380.

 

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/aeidon/default.nix [hunk 1]
```diff
@@ -13,15 +14,17 @@ buildPythonPackage rec {
...
     pname = "aeidon";
     inherit version;
-    sha256 = "sha256-qGpGraRZFVaW1Jys24qvfPo5WDg7Q/fhvm44JH8ulVw=";
+    hash = "sha256-qGpGraRZFVaW1Jys24qvfPo5WDg7Q/fhvm44JH8ulVw=";
   };
 
-  nativeBuildInputs = [
+  build-system = [
+    setuptools
     gettext
```
Comment (dotlambda) [inline pkgs/development/python-modules/aeidon/default.nix:27]: That's not the build-system.

File: pkgs/development/python-modules/aeidon/default.nix [hunk 2]
```diff
@@ -35,11 +38,14 @@ buildPythonPackage rec {
   nativeCheckInputs = [
     pytestCheckHook
     charset-normalizer
+    flake8
```
Comment (dotlambda) [inline pkgs/development/python-modules/aeidon/default.nix:41]: We don't want to check that.

**Review by dotlambda — COMMENTED** [review]
You didn't remove borgbackup from dependencies.

**Comment by erictapen** [comment]
> You didn't remove borgbackup from dependencies.

See https://github.com/NixOS/nixpkgs/pull/325541#discussion_r1736670633

File: pkgs/development/python-modules/aeidon/default.nix (hunk 2 above)
Comment (erictapen) [reply pkgs/development/python-modules/aeidon/default.nix:41]: Why not?

File: pkgs/development/python-modules/aeidon/default.nix (hunk 1 above)
Comment (erictapen) [reply pkgs/development/python-modules/aeidon/default.nix:27]: Though it's required at build time. Should I move it into nativeBuildInputs then?

**Review by fricklerhandwerk — APPROVED** [review]
@erictapen Thanks for following up on the review notes

File: pkgs/development/python-modules/django-crispy-bootstrap3/default.nix
Comment (GetPsyched) [inline pkgs/development/python-modules/django-crispy-bootstrap3/default.nix:1]:
```nix
changelog = "https://github.com/django-crispy-forms/crispy-bootstrap3/blob/${version}/CHANGELOG.md";
```

File: pkgs/development/python-modules/ahocorasick-rs/default.nix
Comment (GetPsyched) [inline pkgs/development/python-modules/ahocorasick-rs/default.nix:1]:
```nix
changelog = "https://github.com/G-Research/ahocorasick_rs/blob/${version}/CHANGELOG.md";
```

**Review by GetPsyched — CHANGES_REQUESTED** [review]
A lot of meta changes :)

File: pkgs/by-name/we/weblate/package.nix [hunk 3]
```diff
@@ -79,7 +79,7 @@ python.pkgs.buildPythonApplication rec {
   dependencies = with python.pkgs; [
     aeidon
     ahocorasick-rs
-    borgbackup
+    ((toPythonModule borgbackup).override { python3 = python; })
```
Comment (dotlambda) [inline pkgs/by-name/we/weblate/package.nix:82]:
```suggestion
    (toPythonModule (borgbackup.override { python3 = python; }))
```

File: pkgs/development/python-modules/translation-finder/default.nix
[the remark this answers is not in the document]
Comment (erictapen) [reply pkgs/development/python-modules/translation-finder/default.nix:1]: Looks like cc0 is only used for Markdown files in the repository or CI config, as in files that don't change the behaviour of the software. Do we really need to set this as the license of the package? From my understanding we don't set the licenses of the source repository but the licenses of the code that constitutes the software.

**Comment by erictapen** [comment]
I went through all the licenses introduced by https://github.com/NixOS/nixpkgs/pull/325541 and https://github.com/NixOS/nixpkgs/pull/334849:

- weblate
- aeidon
- ahocorasick-rs
- celery
- django-crispy-bootstrap3
- phply
- pyicumessageformat
- siphashc
- translate-toolkit
- translation-finder
- weblate-language-data
- weblate-schemas
- cyrtranslit
- django-otp-webauthn

I actually had to fix the license for `siphashc` and extend the one for `weblate` itself, as some Python files use MIT.

Weblate vendors in a few dependencies with licenses such as `Apache-2.0` and `BSD-3-Clause`, do we mention vendored dependencies in `licenses`?

As laid out in https://github.com/NixOS/nixpkgs/pull/338217#discussion_r1740646403, I'd leave CC0 licenses in files like `SECURITY.md` or CI configuration off for now. Or what do you think?

Will squash commits later when approved.

File: pkgs/development/python-modules/translation-finder/default.nix
[the remark this answers is not in the document]
Comment (GetPsyched) [reply pkgs/development/python-modules/translation-finder/default.nix:1]: Oh, hmm; the CI is irrelevant to the source, so it makes sense not to include it. I'd say ignore the license change then.

**Pull request merged**